In [1]:
import pandas as pd
import numpy as np

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [ ]:
# demo = "../data/raw/df_final_demo.txt"
# webdata1 = "../data/raw/df_final_web_data_pt_1.txt"
# webdata2 = "../data/raw/df_final_web_data_pt_2.txt"
# experiment = "../data/raw/df_final_experiment_clients.txt"

# df_demo = pd.read_csv(demo)
# df_webdata1 = pd.read_csv(webdata1)
# df_webdata2 = pd.read_csv(webdata2)
# df_exp = pd.read_csv(experiment)

In [4]:
df_exp = pd.read_csv('/Users/sashacrowe/codebase-ih/vanguard-ab-test/data/raw/df_final_experiment_clients.txt')

## First Look at the Dataset

In [5]:
df_exp.head()

client_id Variation
0    9988021      Test
1    8320017      Test
2    4033851   Control
3    1982004      Test
4    9294070   Control

In [6]:
df_exp.shape

(70609, 2)

In [7]:
df_exp.dtypes

client_id     int64
Variation    object
dtype: object

**DATA BREAKDOWN**    
Column name changes -> Variation should be in lowercase
- Variation (object:string) ->  tells us who was in a test group vs who was in a control group

In [9]:
# Checking for missing data
df_exp.isnull().sum().sort_values(ascending=False)

Variation    20109
client_id        0
dtype: int64

In [12]:
round((20109/df_exp.shape[0])*100,2)

28.48

In [17]:
df_exp['Variation'].value_counts()

Variation
Test       26968
Control    23532
Name: count, dtype: int64

In [21]:
df_exp['Variation'].value_counts(normalize=True)*100

Variation
Test       53.40198
Control    46.59802
Name: proportion, dtype: float64

## What needs cleaning
- the number of rows matches with the client_profiles dataset (70609 records each)
- for the null values in variation we should fill it with value 'untested'
- close to 30% of our clients were part of the A/B test... of which 53% were tested & 47% were control

## A/B TEST INFO
An A/B test was set into motion from 3/15/2017 to 6/20/2017 by the team.

- Control Group: Clients interacted with Vanguard’s traditional online process.
- Test Group: Clients experienced the new, spruced-up digital interface.   

Both groups navigated through an identical process sequence: an initial page, three subsequent steps, and finally, a confirmation page signaling process completion.

The goal is to see if the new design leads to a better user experience and higher process completion rates.

